File: ALODWebUtility.CS/Perms/OnlineUserList.cs
using System;
using System.Data;
using ALOD.Data;
using ALODWebUtility.DataAccess;


namespace ALODWebUtility.Perms
{
    public class OnlineUserList
    {
        private DataSets.OnlineUsersDataTable _users;

        public DataSets.OnlineUsersDataTable GetUsers()
        {
            _users = new DataSets.OnlineUsersDataTable();
            SqlDataStore adapter = new SqlDataStore();
            adapter.ExecuteReader(UserReader, "core_user_sp_GetUsersOnline");
            return _users;
        }

        private void UserReader(SqlDataStore adapter, IDataReader reader)
        {
            // 0-userid, 1-userName, 2-roleName, 3-regionName, 4-loginTime
            DataSets.OnlineUsersRow row = _users.NewOnlineUsersRow();

            row.userId = adapter.GetInteger(reader, 0);
            row.userName = adapter.GetString(reader, 1);
            row.roleName = adapter.GetString(reader, 2);
            row.regionName = adapter.GetString(reader, 3);
            row.loginTime = adapter.GetDateTime(reader, 4);
            row.timeOnline = DateTime.Now.Subtract(row.loginTime);

            _users.Rows.Add(row);
        }
    }
}

File: ALODWebUtility/Perms/OnlineUserList.vb
Imports ALOD.Data

Namespace Permission

    Public Class OnlineUserList

        Private _users As DataSets.OnlineUsersDataTable

        Public Function GetUsers() As DataSets.OnlineUsersDataTable

            _users = New DataSets.OnlineUsersDataTable()
            Dim adapter As New SqlDataStore()
            adapter.ExecuteReader(AddressOf UserReader, "core_user_sp_GetUsersOnline")
            Return _users

        End Function

        Private Sub UserReader(ByVal adapter As SqlDataStore, ByVal reader As IDataReader)

            '0-userid, 1-userName, 2-roleName, 3-regionName, 4-loginTime
            Dim row As DataSets.OnlineUsersRow = _users.NewOnlineUsersRow()

            row.userId = adapter.GetInteger(reader, 0)
            row.userName = adapter.GetString(reader, 1)
            row.roleName = adapter.GetString(reader, 2)
            row.regionName = adapter.GetString(reader, 3)
            row.loginTime = adapter.GetDateTime(reader, 4)
            row.timeOnline = Now.Subtract(row.loginTime)

            _users.Rows.Add(row)

        End Sub

    End Class

End Namespace